nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -440,12 +539,12 @@ public static DataType inferDataType(final Object value, final DataType defaultT
 //            final DataType elementDataType = inferDataType(valueFromMap, RecordFieldType.STRING.getDataType());
 //            return RecordFieldType.MAP.getMapDataType(elementDataType);
         }
-        if (value instanceof Object[]) {
-            final Object[] array = (Object[]) value;
-
+        if (value.getClass().isArray()) {
```

Comment:
Neat! I don't think I realized that `Class.isArray()` and `Array.getLength()` / `Array.get()` were a thing.